### PR TITLE
ensure ids are unique when rendering multiple tables

### PR DESCRIPTION
**Pull request by angus-88**
- auto generated the unique table id if not passed regardless of other options
- added tableId to row and cell id prop to ensure ids are unique if there are multiple tables in a page

### CODE REVIEW

**Comment by wdh2100** [comment]
@angus-88  thank pr!

> added tableId to row and cell id prop to ensure ids are unique if there are multiple tables in a page

Is it the case of calling the mui components at the webpage?

If so, what's wrong with the `ID`?

**Comment by angus-88** [comment]
Hi @wdh2100 

The ids passed to the rows and cell components is a hard coded string combined with an index based on the row or column index. This means for pages with multiple `MUI DataTable` components the Ids would clash.

I have simply added the unique `tableId` prop to the generated ids to ensure uniqueness of ids when rendering multiple tables.

**Comment by wdh2100** [comment]
> Hi @wdh2100
> 
> The ids passed to the rows and cell components is a hard coded string combined with an index based on the row or column index. This means for pages with multiple `MUI DataTable` components the Ids would clash.
> 
> I have simply added the unique `tableId` prop to the generated ids to ensure uniqueness of ids when rendering multiple tables.

@angus-88 

I understood the `uniqueness`.

File: src/MUIDataTable.js [hunk 1]
```diff
@@ -359,8 +359,8 @@ class MUIDataTable extends React.Component {
       props.options.selectToolbarPlacement = STP.NONE;
     }
 
-    // provide default tableId when draggableColumns is enabled and no tableId has been passed as prop
-    if (props.options.draggableColumns && props.options.draggableColumns.enabled === true && !props.options.tableId) {
+    // provide default tableId when no tableId has been passed as prop
+    if (!props.options.tableId) {
```
[the remark this answers is not in the document]
Comment (angus-88) [reply src/MUIDataTable.js:363]: yes exactly

**Comment by DrNguyen2525** [comment]
Hi. I also saw DOM alert about non-unique ID of pagination buttons. More details here #1519.
Any one have a solution ?

**Comment by angus-88** [comment]
Is this ready to merge then what's the hold up, if not what's outstanding so I can sort it out?

**Comment by DrNguyen2525** [comment]
@gregnb

**Review by wdh2100 — APPROVED** [review]
LGTM 👍

**Pull request merged**